tests/components/modbus/test_init.py

```diff
@@ -703,34 +703,33 @@ async def test_delay(hass, mock_pymodbus):
         ]
     }
     mock_pymodbus.read_coils.return_value = ReadResult([0x01])
-    now = dt_util.utcnow()
-    with mock.patch("homeassistant.helpers.event.dt_util.utcnow", return_value=now):
-        assert await async_setup_component(hass, DOMAIN, config) is True
-        await hass.async_block_till_done()
-
-    # pass first scan_interval
-    start_time = now
-    now = now + timedelta(seconds=(set_scan_interval + 1))
+    start_time = dt_util.utcnow()
     with mock.patch(
-        "homeassistant.helpers.event.dt_util.utcnow", return_value=now, autospec=True
+        "homeassistant.helpers.event.dt_util.utcnow", return_value=start_time
```

Comment:
I think that in general it is not good to patch `utcnow` since it may affect other states

Reply:
I believe it is the only way to simulate timing, but I am open for other suggestions. For the record there are at least 243 other lines in other components that do the same thing, so it cannot be that bad.

I see 2 alternatives, but both with heavy side effects:
a) let the time run physically and do simple sleeps.
This would mean this test would take 26seconds instead of 120milli seconds, which is clearly unacceptable.

b) Do not patch utcnow, but continue calling async_fire_time_changed(hass, now).
This will really cause problems in core, because some functions use utcnow. so patching utcnow keep core in sync.

c) not test keywords like "delay"
that of course is not a good thing.

There are parts of core that are affected when you patch utcnow, everything that starts a timer will be terminated differently (I suppose e.g. discovery would be affected), but the state machine is as far as I can debug unaffected.

Reply:
But as I started by writing, I am all ears for other solutions.